### ISSUE TITLE
Version 4.0.1: NativeEngine is lacking delegate calls

### ISSUE THREAD

**Issue by tfonfara** [daltoniam/Starscream#757]
After updating to version 4.0.1 with iOS 13 the new NativeEngine is used. I noticed that at least (maybe even more) the .connected event isn't sent even if the websocket did connect.
Workaround for now is to set useCustomEngine to true in order to prevent using the new engine. I would suggest for the future to use a version number like 4.1.0 if major changes like that are being introduced.

**Comment by kylekurz** [comment]
IMO, this is worse than using a bugfix semver, for a major change. This is doing that with a feature that isn't complete. If NativeEngine is the default, it should be a major revision (e.g. 5.0.0) *and* it should be either well-documented with a new delegate API or it should be fully compliant with the old API. As it is, bugfix upgrades are now broken if they rely on any of the delegate methods you've not implemented in NativeEngine.

**Comment by kylekurz** [comment]
FWIW, a more thorough test infrastructure (not just "will it build") would have caught this. I'm happy to help out as available with some improvements, but I'd really suggest a couple things:

1. New PRs from outside *must* include test coverage and pass existing tests.
2. Major architectural changes should come with new test coverage and you can use tests you need to modify to provide release notes about what upgrading users need to adjust.

Just a little more thought on this.

**Comment by daltoniam** [comment]
@tfonfara I agree, I totally overlooked that in the release, my bad. I set useCustomEngine to true by default now as the native WS APIs don't appear to have all the features Starscream provides. I released 4.0.2 with this update to fix the issue.

@kylekurz I agree testing infrastructure would be great, but I'm only able to work on Starscream in my free time which is limited. I started on that with the unit tests of implementing all the autobahn fuzzing server tests and there is an outstanding issue (#558) to add better testing infrastructure. PR is welcomed.
